### ISSUE TITLE
village/campus ~ Woodstock

### ISSUE THREAD

**Issue by AdrianKriger** [AdrianKriger/osm_LoD1_3DCityModel#6]
Woodstock ~ Relation: [(2034285)](https://www.openstreetmap.org/relation/2034285#map=14/-33.9300/18.4500) fails?

**Comment by AdrianKriger** [comment]
Add [Cape Town City Center](https://www.openstreetmap.org/relation/2034619#map=14/-33.9220/18.4235) -> to village/campus -> as well. This might be why [Cape Town Ward 115](https://www.openstreetmap.org/relation/6580101#map=13/-33.9169/18.4360) -> districts -> fails.

Mouille Point, V&A Waterfront, Foreshore -> (village/campus) -> and Ward 57 -> (district) -> pass.

Definitely some feature in the area. Try to isolate.

**Comment by AdrianKriger** [comment]
Fixed with [07b23d0](https://github.com/AdrianKriger/osm_LoD1_3DCityModel/commit/07b23d09c4b22afb36b5378efe5444e18dfe8b5f#diff-17108199f44b0e70c80f1259c285accd8367ded8f50e20010b2321e102e27abd)

With reference to [village/campus](https://github.com/AdrianKriger/osm_LoD1_3DCityModel/tree/main/village_campus):
a) `Linestring` buildings that are not polygons
```python
 # take care of non-Polygon LineString's 
    for i, row in ts.iterrows():
        if row.geometry.type == 'LineString' and len(row.geometry.coords) < 3:
            ts = ts.drop(ts.index[i])
```
b) aoi relations that return many areas
```python
# when relations return may areas
    if b_type == 'relation' and len(aoi) > 1:
        for i, row in aoi.iterrows():
            if row.tags != None and 'place' in row.tags:
                focus = row
            
        trim = pd.DataFrame(focus)
        trim = trim.T
        aoi = gpd.GeoDataFrame(trim, geometry = trim['geometry'])
```
I'll open a separate issue for [Districts](https://github.com/AdrianKriger/osm_LoD1_3DCityModel/tree/main/districts).
